pkgs/top-level/all-packages.nix

```diff
@@ -19766,7 +19766,9 @@ with pkgs;
   valgrind = callPackage ../development/tools/analysis/valgrind {
     inherit (buildPackages.darwin) xnu bootstrap_cmds cctools;
   };
-  valgrind-light = res.valgrind.override { gdb = null; };
+  valgrind-light = (res.valgrind.override { gdb = null; }).overrideAttrs (oldAttrs: {
+    meta.description = "${oldAttrs.meta.description} (without GDB)";
```

Comment:
Unfortunately this drops other `meta` fields: 

```
nix-repl> :p builtins.attrNames legacyPackages.x86_64-linux.valgrind.meta       
[ "available" "badPlatforms" "broken" "description" "homepage" "insecure" "license" "longDescription" "maintainers" "name" "outputsToInstall" "platforms" "position" "unfree" "unsupported" ]

nix-repl> :p builtins.attrNames legacyPackages.x86_64-linux.valgrind-light.meta
[ "available" "broken" "description" "insecure" "name" "outputsToInstall" "position" "unfree" "unsupported" ]
```

this should probably be somethling like

```nix
meta = oldAttrs.meta // { description = ...; };
```